src/diffusers/training_utils.py

```diff
@@ -66,7 +68,12 @@ def __init__(
             power (float): Exponential factor of EMA warmup. Default: 2/3.
             min_value (float): The minimum EMA decay rate. Default: 0.
         """
-
+        deprecation_message = (
```

Comment:
Ah I didn't know that we already had a class here.

There is a relatively high chance that the EMA model will be moved here eventually if other training scripts use it. Should we maybe just update it now with a breaking change as it would be cleaner? If not:

- We have a deprecated EMAModel in `src/diffusers` that some people still use
- Two EMAModel in examples
- A big chance that we need to overwrite the deprecated model with the example one soon

=> Maybe easier to just do a breaking change now? Wdyt @patil-suraj @anton-l 

Not super happy about the two classes situation

Reply:
Ping here @anton-l @patil-suraj

Reply:
@patil-suraj feel free to take over if that's the case! I've tried to make unconditional EMAModel behave similarly to the one you've added to text2img. However, I'm not sure how easy it is to replace the integrated EMAModel without breaking changes: it has a slightly different API to the script counterparts

Reply:
Awesome, Will try to see if we can support both EMA warmup schedules cleanly.

Reply:
Will push the changes in this PR.